File: sources/academy/webscraping/scraping_basics_javascript/exercises/warehouse_units_regex.mjs
import * as cheerio from 'cheerio';

function parseUnitsText(text) {
  const match = text.match(/\d+/);
  if (match) {
    return Number.parseInt(match[0], 10);
  }
  return 0;
}

const url = 'https://warehouse-theme-metal.myshopify.com/collections/sales';
const response = await fetch(url);

if (!response.ok) {
  throw new Error(`HTTP ${response.status}`);
}

const html = await response.text();
const $ = cheerio.load(html);

for (const element of $('.product-item').toArray()) {
  const $productItem = $(element);

  const $title = $productItem.find('.product-item__title');
  const title = $title.text().trim();

  const unitsText = $productItem.find('.product-item__inventory').text();
  const unitsCount = parseUnitsText(unitsText);

  console.log(`${title} | ${unitsCount}`);
}

File: sources/academy/webscraping/scraping_basics_python/exercises/warehouse_units_regex.py
import re

import httpx
from bs4 import BeautifulSoup

url = "https://warehouse-theme-metal.myshopify.com/collections/sales"
response = httpx.get(url)
response.raise_for_status()

soup = BeautifulSoup(response.text, "html.parser")

for product in soup.select('.product-item'):
    title = product.select_one('.product-item__title').text.strip()

    units_text = product.select_one('.product-item__inventory').text
    match = re.search(r"\d+", units_text)
    units = int(match.group()) if match else 0

    print(f"{title} | {units}")
